frontend/workflows/projectSelector/src/types.tsx

```diff
@@ -37,27 +37,70 @@ interface BackgroundPayload {
 
 export type Action = BackgroundAction | UserAction;
 
-export interface State {
+export interface ProjectsState {
   [Group.PROJECTS]: GroupState;
   [Group.UPSTREAM]: GroupState;
   [Group.DOWNSTREAM]: GroupState;
-
-  projectData: { [projectName: string]: IClutch.core.project.v1.IProject };
-  loading: boolean;
-  error: ClutchError | undefined;
 }
 
-interface GroupState {
+export interface GroupState {
   [projectName: string]: ProjectState;
 }
 
+// n.b. if you are updating ProjectState be sure to update the custom type guard below it.
 export interface ProjectState {
   checked: boolean;
   // TODO: hidden should be derived?
   hidden?: boolean; // upstreams and downstreams are hidden when their parent is unchecked unless other parents also use them.
   custom?: boolean;
 }
 
+/**
+ * Determines if an object is of type @type {ProjectState}.
+ */
+const isProjectState = (state: ProjectState | object): state is ProjectState => {
+  const pState = state as ProjectState;
+  const checkedProp = pState?.checked;
+  const hasRequiredProps = checkedProp !== undefined && typeof checkedProp === "boolean";
+
+  const validOptionalProps =
+    (pState?.hidden !== undefined ? typeof pState.hidden === "boolean" : true) &&
```

Comment:
delete anything related to hidden, it's not used, it's derived instead of stored on the state